airflow/providers/SUSPENDING_AND_RESUMING_PROVIDERS.rst

```diff
@@ -0,0 +1,150 @@
+        from airflow.providers.google.cloud.hooks.dataflow import (
+    airflow/providers/google/cloud/hooks/dataflow.py:32: in <module>
+        from google.cloud.dataflow_v1beta3 import GetJobRequest, Job, JobState, JobsV1Beta3AsyncClient, JobView
+    E   ModuleNotFoundError: No module named 'google.cloud.dataflow_v1beta3'
+    _ ERROR collecting tests/providers/microsoft/azure/transfers/test_azure_blob_to_gcs.py _
+
+
+The fix is to add this line at the top of the ``tests/providers/apache/beam/operators/test_beam.py`` module:
+
+  .. code-block:: python
+
+    pytest.importorskip("apache.airflow.providers.google")
+
+
+* Some of the other providers might also just import unconditionally the suspended provider and they will
+  fail during provider verification step in CI. In this case you should turn the provider imports
+  into conditional imports. For example when import fails after ``amazon`` provider has been suspended:
+
+  .. code-block:: txt
+
+      Traceback (most recent call last):
+        File "/opt/airflow/scripts/in_container/verify_providers.py", line 266, in import_all_classes
+          _module = importlib.import_module(modinfo.name)
+        File "/usr/local/lib/python3.7/importlib/__init__.py", line 127, in import_module
+          return _bootstrap._gcd_import(name, package, level)
+        File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
+        File "<frozen importlib._bootstrap>", line 983, in _find_and_load
+        File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
+        File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
+        File "<frozen importlib._bootstrap_external>", line 728, in exec_module
+        File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
+        File "/usr/local/lib/python3.7/site-packages/airflow/providers/mysql/transfers/s3_to_mysql.py", line 23, in <module>
+          from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+      ModuleNotFoundError: No module named 'airflow.providers.amazon'
+
+or:
+
+  .. code-block:: txt
+
+  Error: The ``airflow.providers.microsoft.azure.transfers.azure_blob_to_gcs`` object in transfers list in
+  airflow/providers/microsoft/azure/provider.yaml does not exist or is not a module:
+  No module named 'gcloud.aio.storage'
+
+The fix for that is to turn the feature into optional provider feature (in the place where the excluded
+``airflow.providers`` import happens:
+
+  .. code-block:: python
+
+    try:
+        from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+    except ImportError as e:
+        from airflow.exceptions import AirflowOptionalProviderFeatureException
+
+        raise AirflowOptionalProviderFeatureException(e)
+
+
+* In case we suspend important provider, which is part of the default Dockerfile or used you might want to
+  update tests for PROD docker image in ``docker_tests/test_prod_image.py``.
```

Comment:
```suggestion
* In case we suspend an important provider, which is part of the default Dockerfile you might want to
  update the tests for PROD docker image in ``docker_tests/test_prod_image.py``.
```